[RFR] Support API empty response on DELETE and DELETE_MANY actions (#3441)
marmelab/react-admin: docs/DataProviders.md

`CREATE`             | `{ data: {Record} }`
`UPDATE`             | `{ data: {Record} }`
`UPDATE_MANY`        | `{ data: {mixed[]} }` The ids which have been updated

> `DELETE`             | `{ data: {Record} }` The resource that had been deleted or nothing

Reviewer: ```suggestion
`DELETE`             | `{ data: {Record} }` The record that has been deleted
```

Author: I would precise, to the data provider maintainers that this data isn't required.

> `DELETE_MANY`        | `{ data: {mixed[]} }` The ids which have been deleted or an empty array

Reviewer: ```suggestion
`DELETE_MANY`        | `{ data: {mixed[]} }` The ids of the deleted records
```

Author: Idem, I would precise, to the data provider maintainers that this data isn't required.

                    };
                case CREATE:
                    return { data: { ...params.data, id: json.id } };
                case DELETE_MANY:

>                     return { data: json ? json : [] };

Reviewer: ```suggestion
                    return { data: json || [] };
```
